src/timer_thread.c

```diff
@@ -0,0 +1,452 @@
+               {"BUS", SIGBUS},
+#endif
+#ifdef SIGSEGV
+               {"SEGV", SIGSEGV},
+#endif
+#ifdef SIGSYS
+               {"SYS", SIGSYS},
+#endif
+#ifdef SIGPIPE
+               {"PIPE", SIGPIPE},
+#endif
+#ifdef SIGALRM
+               {"ALRM", SIGALRM},
+#endif
+#ifdef SIGTERM
+               {"TERM", SIGTERM},
+#endif
+#ifdef SIGURG
+               {"URG", SIGURG},
+#endif
+#ifdef SIGSTOP
+               {"STOP", SIGSTOP},
+#endif
+#ifdef SIGTSTP
+               {"TSTP", SIGTSTP},
+#endif
+#ifdef SIGCONT
+               {"CONT", SIGCONT},
+#endif
+#ifdef SIGCHLD
+               {"CHLD", SIGCHLD},
+#endif
+#ifdef SIGCLD
+               {"CLD", SIGCLD},
+#else
+#ifdef SIGCHLD
+               {"CLD", SIGCHLD},
+#endif
+#endif
+#ifdef SIGTTIN
+               {"TTIN", SIGTTIN},
+#endif
+#ifdef SIGTTOU
+               {"TTOU", SIGTTOU},
+#endif
+#ifdef SIGIO
+               {"IO", SIGIO},
+#endif
+#ifdef SIGXCPU
+               {"XCPU", SIGXCPU},
+#endif
+#ifdef SIGXFSZ
+               {"XFSZ", SIGXFSZ},
+#endif
+#ifdef SIGVTALRM
+               {"VTALRM", SIGVTALRM},
+#endif
+#ifdef SIGPROF
+               {"PROF", SIGPROF},
+#endif
+#ifdef SIGWINCH
+               {"WINCH", SIGWINCH},
+#endif
+#ifdef SIGUSR1
+               {"USR1", SIGUSR1},
+#endif
+#ifdef SIGUSR2
+               {"USR2", SIGUSR2},
+#endif
+#ifdef SIGLOST
+               {"LOST", SIGLOST},
+#endif
+#ifdef SIGMSG
+               {"MSG", SIGMSG},
+#endif
+#ifdef SIGPWR
+               {"PWR", SIGPWR},
+#endif
+#ifdef SIGPOLL
+               {"POLL", SIGPOLL},
+#endif
+#ifdef SIGDANGER
+               {"DANGER", SIGDANGER},
+#endif
+#ifdef SIGMIGRATE
+               {"MIGRATE", SIGMIGRATE},
+#endif
+#ifdef SIGPRE
+               {"PRE", SIGPRE},
+#endif
+#ifdef SIGGRANT
+               {"GRANT", SIGGRANT},
+#endif
+#ifdef SIGRETRACT
+               {"RETRACT", SIGRETRACT},
+#endif
+#ifdef SIGSOUND
+               {"SOUND", SIGSOUND},
+#endif
+#ifdef SIGINFO
+               {"INFO", SIGINFO},
+#endif
+               {NULL, 0}};
+
+static int signm2signo(const char *nm)
+{
+  const struct signals *sigs;
+
+  for (sigs = siglist; sigs->signm; sigs++) {
+    if (strcmp(sigs->signm, nm) == 0)
+      return sigs->signo;
+  }
+
+  /* Handle RT Signal#0 as special for strtol's err spec */
+  if (strcmp("RT0", nm) == 0)
+    return SIGRTMIN;
+
+  if (strncmp("RT", nm, 2) == 0) {
+    int ret = (int)strtol(nm + 2, NULL, 0);
+    if (!ret || (SIGRTMIN + ret > SIGRTMAX))
+      return 0;
+    return SIGRTMIN + ret;
+  }
+  return 0;
+}
+
+static int mrb_to_signo(mrb_state *mrb, mrb_value vsig)
+{
+  int sig = -1;
+  const char *s;
+
+  switch (mrb_type(vsig)) {
+  case MRB_TT_FIXNUM:
+    sig = mrb_fixnum(vsig);
+    if (sig < 0 || sig >= SIGRTMAX) {
+      mrb_raisef(mrb, E_ARGUMENT_ERROR, "invalid signal number (%S)", vsig);
+    }
+    break;
+  case MRB_TT_SYMBOL:
+    s = mrb_sym2name(mrb, mrb_symbol(vsig));
+    if (!s)
+      mrb_raise(mrb, E_ARGUMENT_ERROR, "bad signal");
```

Comment:
It is better to unify it to either `if (hoge) fuga;` or `if (hoge) { fuga; }` notation.